improver/cli/threshold.py

```diff
@@ -140,29 +140,32 @@ def process(
         thresholds = [np.float32(x) for x in threshold_values]
         fuzzy_bounds = None
 
-    result_no_collapse_coord = BasicThreshold(
+    each_threshold_func_list = []
+
+    if vicinity is not None:
+        # smooth thresholded occurrences over local vicinity
+        each_threshold_func_list.append(OccurrenceWithinVicinity(vicinity))
+
+    if collapse_coord is not None:
+        # Take a weighted mean across realizations with equal weights
+        each_threshold_func_list.append(
+            WeightAndBlend(collapse_coord, "linear", y0val=1.0, ynval=1.0)
+        )
+
+    result = BasicThreshold(
         thresholds,
         fuzzy_factor=fuzzy_factor,
         fuzzy_bounds=fuzzy_bounds,
         threshold_units=threshold_units,
         comparison_operator=comparison_operator,
+        each_threshold_func=each_threshold_func_list,
     )(cube)
 
     if vicinity is not None:
-        # smooth thresholded occurrences over local vicinity
-        result_no_collapse_coord = OccurrenceWithinVicinity(vicinity)(
-            result_no_collapse_coord
-        )
-        new_cube_name = in_vicinity_name_format(result_no_collapse_coord.name())
-        result_no_collapse_coord.rename(new_cube_name)
-
-    if collapse_coord is None:
-        return result_no_collapse_coord
+        result.rename(in_vicinity_name_format(result.name()))
 
     # Raise warning if result_no_collapse_coord is masked array
-    if np.ma.isMaskedArray(result_no_collapse_coord.data):
-        warnings.warn("Collapse-coord option not fully tested with " "masked data.")
-    # Take a weighted mean across realizations with equal weights
-    plugin = WeightAndBlend(collapse_coord, "linear", y0val=1.0, ynval=1.0)
+    if collapse_coord is not None and np.ma.isMaskedArray(result.data):
+        warnings.warn("Collapse-coord option not fully tested with masked data.")
```

Comment:
It's about time we did test this given we're collapsing cloud data - I'll check with the relevant theme owner(s).

Reply:
follow on PR?

Reply:
Yes that's what I was implying.  I'll put it on my list to follow this up.